File: 8/fawn_wong/picmaker.py
import os, sys, math

pic = open("pic.ppm", "w")
pic.write("P3 \n")
pic.write("500 500 \n")
pic.write("255 \n")

for x in range(500):
    for y in range(500):
        if (2 * x + 3 * y) % 500 < 300:
            r = 0
            g = x + y % 255
            b = x + y % 255
        else:
            r = 255
            g = 255
            b = 255
        
        pic.write("%d %d %d "%(r, g, b))
    pic.write("\n")

pic.close()    

File: 4/shahruz_ghaemi/picmaker.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <fcntl.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <string.h>
#include <unistd.h>

int main(){
  int fd, i, j, r, g, b;

  fd = open("pic.ppm", O_CREAT | O_WRONLY | O_TRUNC, 0664);
  char line[256];

  write(fd,"P3 500 500 255 ",16);

  for(i = 0; i<500; i++){
    for(j = 0; j<500; j++){
      
      r = 255;
      g = j%255;
      b = i%255;
      sprintf(line, " %d %d %d", r, g, b);
      write(fd, line, strlen(line));
    }
  }
  close(fd);
  return 0;
}
